src/eth/transaction.rs

```diff
@@ -223,11 +280,209 @@ pub fn format_address(value: &Option<EthAddress>) -> BytesMut {
     }
 }
 
+/// Pads data with leading zeros to the specified length
+pub fn pad_leading_zeros(data: &[u8], length: usize) -> Vec<u8> {
+    if data.len() >= length {
+        return data.to_vec();
+    }
+    let mut zeros = vec![0; length - data.len()];
+    zeros.extend_from_slice(data);
+    zeros
+}
+
+/// Parses raw bytes into an eth transaction
+pub fn parse_eth_transaction(data: &[u8]) -> anyhow::Result<EthTx> {
+    ensure!(!data.is_empty(), "eth transaction data is empty");
+
+    match *data.first().context("failed to get signature prefix")? as u64 {
```

Comment:
This error handling can be omitted with construction like `match data { Some(EIP_1559_TX_TYPE) => ... `, making the code simpler.

Reply:
fixed